src/main/java/svydovets/core/context/DefaultApplicationContext.java

```diff
@@ -97,8 +99,44 @@ private Object initWithBeanPostProcessor(Map.Entry<String, BeanDefinition> entry
         return postProcessAfterInitialization(bean, beanName);
     }
 
+    /**
+     * Invokes methods that were annotated with @PostConstruct annotation.
+     */
+    private void invokePostConstructMethods() {
+        beanMap.values()
+                .forEach(this::postConstructInitialization);
+    }
+
+    /**
+     * Invokes the founded method with annotation @PostConstruct. If the class has more than one annotated method,
+     * the method will throw exception NoUniquePostConstructException.
+     * @param bean represents the initialized bean
+     * @throws InvalidInvokePostConstructMethodException throws this exception if something went wrong
+     */
     private void postConstructInitialization(Object bean) {
-        throw new UnsupportedOperationException();
+        Class<?> beanType = bean.getClass();
+        Method[] declaredMethods = beanType.getDeclaredMethods();
+        Predicate<Method> isAnnotatedMethod = method -> method.isAnnotationPresent(PostConstruct.class);
+
+        var filteredMethods = Arrays.stream(declaredMethods).filter(isAnnotatedMethod);
+
+        boolean isNotUniqueMethod = filteredMethods.count() > 1;
```

Comment:
You can simplify it:
boolean isNotUniqueMethod = Arrays.stream(declaredMethods)
.filter(method -> method.isAnnotationPresent(PostConstruct.class))
.count() > 1

Reply:
Fixed in the PR https://github.com/Svydovets-Bobocode-Java-Ultimate-3-0/Bring/pull/10